lib/v2/reuters/common.js

```diff
@@ -9,31 +9,54 @@ module.exports = async (ctx) => {
     const topic = ctx.params.topic ?? (category === 'authors' ? 'reuters' : '');
     const limit = ctx.query.limit ? parseInt(ctx.query.limit) : 20;
 
-    const rootUrl = 'https://www.reuters.com';
-    const currentUrl = topic ? `${rootUrl}/${category}/${topic}/` : `${rootUrl}/${category}/`;
-    const response = await got(currentUrl);
-    const $ = cheerio.load(response.data);
-
-    let items = $('.media-story-card__body__3tRWy a.media-story-card__heading__eqhp9, a.svelte-pxbp38, a.svelte-11dknnx, a.svelte-e21rsn')
-        .slice(0, limit)
-        .toArray()
-        .map((item) => {
-            item = $(item);
-            item.find('span.visually-hidden__hidden__2qXMW').remove();
+    const MUST_FETCH_BY_TOPICS = ['authors'];
+    const section_id = `/${category}/${topic ? `${topic}/` : ''}`;
+    const { title, description, rootUrl, response } = await (async () => {
+        if (!MUST_FETCH_BY_TOPICS.includes(category)) {
+            const rootUrl = 'https://www.reuters.com/pf/api/v3/content/fetch/articles-by-section-alias-or-id-v1';
+            const response = await got(
+                `${rootUrl}?query={${encodeURI(
+                    JSON.stringify({
+                        offset: 0,
+                        size: limit,
+                        section_id,
+                        website: 'reuters',
+                    }).slice(1, -1)
+                )}}`
+            ).json();
             return {
-                title: item.text(),
-                link: new URL(item.prop('href'), rootUrl).href,
+                title: response.result.section.title,
+                description: response.result.section.section_about,
+                rootUrl,
+                response,
             };
-        });
-    if (!items.length) {
-        const metadata = $('script#fusion-metadata').html();
-        const metadataObj = JSON.parse(metadata.match(/Fusion.globalContent=(\{[\s\S]*?});/)[1]);
-        const articles = metadataObj.arcResult?.articles ?? metadataObj.result?.articles ?? [];
-        items = articles.map((article) => ({
-            title: article.title,
-            link: rootUrl + article.canonical_url,
-        }));
-    }
+        } else {
+            const rootUrl = 'https://www.reuters.com/pf/api/v3/content/fetch/articles-by-topic-v1';
+            const response = await got(
+                `${rootUrl}?query={${encodeURI(
+                    JSON.stringify({
+                        offset: 0,
+                        size: limit,
+                        topic_url: section_id,
+                        website: 'reuters',
+                    }).slice(1, -1)
+                )}}`
```

Comment:
Please use the [searchParams option](https://github.com/sindresorhus/got/tree/v11.8.6#url).